### ISSUE TITLE
Add recipe process condition to check whether a specific FTB quest is complete

### ISSUE THREAD

**Issue by James103** [AztechMC/Modern-Industrialization#514]
Currently, there is no dedicated recipe process condition which checks whether the player (or team) that placed the machine has completed a specific quest or not.

Using a custom process condition to check whether the player that placed the machine has completed a given quest is slow (as it has to run JavaScript code, get the player that placed the machine by its UUID and check whether that player has completed the quest, all once per tick).

It may be possible to write a dedicated machine process condition for this use case which is more optimized, as quest completions, especially for a single player (or team), usually happen much less frequently than once per tick.

Example use cases:
* Certain quests could unlock better recipes.
* Additional requirements to craft the Replicator; an example being 100% completion of all prior quests.
* Requiring the player to craft certain items first before crafting other (higher tier) items; an example being that players can't craft MV circuits unless they craft an LV circuit themselves (trading for it doesn't count).

**Comment by Technici4n** [comment]
Yes this is a good idea. In fact, we already had plans for this. :)

**Comment by James103** [comment]
It may also be wise to support checking for vanilla advancements in this or another dedicated process condition, for mod packs that can't or don't want to use FTB Quests.

**Comment by Technici4n** [comment]
Yes that's a good idea. Would be a completely different condition though. 😉
